### PR TITLE
WIP #7498: add php interactive shell command

### PR DESCRIPTION
**Pull request by pana1990**
WIP : https://github.com/yiisoft/yii2/issues/7498

Task : 
- [ ] Add phpdocs
- [ ] Add @since
- [ ]  Update changelog
- [ ] Add another casters (Suggestion?)

### CODE REVIEW

**Comment by pana1990** [comment]
Example : 

screenshot_3

File: framework/console/controllers/TinkerController.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public function actionIndex()
+    {
+        $config = new Configuration;
+
+        $config->getPresenter()->addCasters(
+            $this->getCasters()
+        );
+        $config->setTabCompletion(true);
```
Comment (bobthecow) [inline framework/console/controllers/TinkerController.php]: This line is unnecessary. Tab completion is enabled by default, and this line will prevent a user from disabling it locally via psysh config.

Comment (pana1990) [inline framework/console/controllers/TinkerController.php]: Yeah, my bad.

Comment (pana1990) [inline framework/console/controllers/TinkerController.php]: @bobthecow  done! thanks for your help :)

Comment (bobthecow) [inline framework/console/controllers/TinkerController.php]: No problem!

File: framework/console/Application.php [hunk 2]
```diff
@@ -184,6 +184,7 @@ public function coreCommands()
             'message' => 'yii\console\controllers\MessageController',
             'migrate' => 'yii\console\controllers\MigrateController',
             'serve' => 'yii\console\controllers\ServeController',
+            'tinker' => 'yii\console\controllers\TinkerController',
```
Comment (samdark) [inline framework/console/Application.php]: What a name :) That's a bit weird.

Comment (pana1990) [inline framework/console/Application.php]: copy/paste from laravel :) 

http://laravel-recipes.com/recipes/280/interacting-with-your-application

Suggestion?

Comment (samdark) [inline framework/console/Application.php]: `shell` as it was in Yii 1.1?

**Comment by samdark** [comment]
Found https://github.com/svkurowski/yii2-shell while googling...

**Comment by pana1990** [comment]
@samdark done! can you include this feature in 2.0.8 milestone?

**Comment by cebe** [comment]
I am usure if we should add a dependency to psysh to the framework. also you now have added it to require-dev which will cause the dependency to not be installed by default.
If may be useful make an extension of this and add this to the application templates by default instead of adding a framework dependency.

**Comment by pana1990** [comment]
@yiisoft/core-developers opinios?

**Comment by klimov-paul** [comment]
I am agree with @cebe.
There should not be dependencies on "psysh" in the core.

We already have a lot of complains about extra dependencies to the external libraries and assets of the core. In 2.1 we are going to split the main repository into several ones in order to solve that. This change just make things worse at this aspect.

This can be released as a separated extension or should be reworked to be more independent.

**Comment by pana1990** [comment]
i am agree, then, can you create repo for this feature?

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2-shell

**Comment by pana1990** [comment]
@samdark thanks.

**Comment by pana1990** [comment]
@samdark i can't fork repo...

**Comment by SilverFire** [comment]
Fixed

**Comment by pana1990** [comment]
i close this pr, new pr in https://github.com/yiisoft/yii2-shell/pulls

**Pull request closed without merging**